mapfishapp/src/main/webapp/app/js/GEOR_Querier.js

```diff
@@ -0,0 +1,321 @@
+
+    /**
+     * Property: layer
+     * {OpenLayers.Layer.Vector} The layer on which geometries can be drawn
+     */
+    layer: null,
+
+    /**
+     * Property: attributeStore
+     * {GeoExt.data.AttributeStore} 
+     */
+    attributeStore: null,
+ 
+    /**
+     * Property: _geometryName
+     * {String} The layer geometry name, obtained by DescribeFeatureType
+     */
+    _geometryName: null,
+
+    /*
+     * Method: initComponent.
+     * Overridden constructor. Set up widgets and lay them out
+     */
+    initComponent: function() {
+        var r = this.record,
+            type = r.get("type"),
+            isWFS = type === "WFS",
+            layer = r.get("layer"), 
+            name = r.get("title") || layer.name || "",
+            geometryName = "";
+
+        this.title = OpenLayers.i18n("Request on NAME", {
+            "NAME": name
+        });
+
+        var pseudoRecord = {
+            typeName: isWFS ? 
+                r.get("WFS_typeName") : r.get("name"),
+            owsURL: isWFS ? 
+                layer.protocol.url : r.get("WFS_URL")
+        };
+
+        GEOR.waiter.show();
+        // get layer model through WFS DescribeFeatureType:
+        this.attributeStore = GEOR.ows.WFSDescribeFeatureType(pseudoRecord, {
+            extractFeatureNS: true,
+            success: function() {
+                // we get the geometry column name, and remove the corresponding record from store
+                var idx = this.attributeStore.find("type", GEOR.ows.matchGeomProperty);
+                if (idx > -1) {
+                    // we have a geometry
+                    var r = this.attributeStore.getAt(idx),
+                        geometryName = r.get("name");
+                    // create the protocol:
+                    this.protocol = GEOR.ows.WFSProtocol(pseudoRecord, this.map, {
+                        geometryName: geometryName
+                    });
+                    this._geometryName = geometryName;
+                    // remove geometry from attribute store:
+                    this.attributeStore.remove(r);
+                } else {
+                    GEOR.util.infoDialog({
+                        msg: tr("querier.layer.no.geom")
+                    });
+                }
+            },
+            failure: function() {
+                GEOR.util.errorDialog({
+                    msg: tr("querier.layer.error")
+                });
+            },
+            scope: this
+        });
+
+        this.layer = new OpenLayers.Layer.Vector("__georchestra_filterbuilder", {
+            displayInLayerSwitcher: false,
+            styleMap: GEOR.util.getStyleMap({
+                "default": {
+                    strokeWidth: 2,
+                    strokeColor: "#ee5400",
+                    fillOpacity: 0
+                }
+            })
+        });
+
+        this.filterbuilder = new Styler.FilterBuilder({
+            defaultBuilderType: Styler.FilterBuilder.ALL_OF,
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["Styler" is not defined.](https://www.codacy.com/app/georchestra/georchestra/file/2338402636/issues/source?bid=3200181&fileBranchId=3209168#l163)